src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -262,6 +268,10 @@ public void validate() throws AppEngineException {
           "Validation Error: Java Tools jar location '"
               + JAR_LOCATIONS.get(JAVA_TOOLS_JAR) + "' is not a file.");
     }
+    if (!Files.isDirectory(getJavaAppEngineComponentsPath())) {
```

Comment:
my concern about having this here is that we can support deploy without needing the java components installed.  A client should be able to get configure gcloud and do a deployment without install the components,